opendatahub-io/odh-dashboard
# handle kserve in global model serving page

Addresses most of #1948 but not everything since the edit modal for kserve isn't yet available. For now the edit action for a kserve inference service is disabled.
Also does not expose GRPC url in table (out of scope).
Does not address deletion of `ServingRuntime` as this will be addressed by `ownerReferences` when we create the kserve `ServingRuntime` and `InferenceService` - related to https://github.com/opendatahub-io/odh-dashboard/issues/1998

## Description


Updates the model serving table to display column for `Serving runtime` and removes the `Model server` column.
Changed title of the global model serving page to `Deployed models`

![image](https://github.com/opendatahub-io/odh-dashboard/assets/14068621/ace16e92-3288-4b06-8c46-c71d32cfef78)

fyi @vconzola 

## How Has This Been Tested?



- Create a model server: `ServingRuntime`
- Deploy a model: `InferenceService`
- Edit the `InferenceService` and change annotation `serving.kserve.io/deploymentMode` to a value other than `ModelMesh`
- Go to the `Model Serving` page using the left navigation
- Observe the changes to the table
- Click the kebab menu for your `InferenceService` and observe that `Edit` is disabled
- Use the `Delete` action
- Observe the `InferenceService` is deleted (Note that until the kserve creation flows are complete, the `ServingRuntime` will not be deleted as we will rely on `ownerReferences`)

## Test Impact


Added some unit tests.

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
